BlueDBOnDisk/src/main/java/io/bluedb/disk/serialization/validation/ObjectValidation.java

```diff
@@ -0,0 +1,68 @@
+package io.bluedb.disk.serialization.validation;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.Modifier;
+import java.util.HashSet;
+import java.util.Set;
+
+public class ObjectValidation {
+	public static void validateFieldValueTypesForObject(Object obj) throws IllegalArgumentException, IllegalAccessException, SerializationException {
+		validateFieldValueTypesForObject(obj, new HashSet<>());
+	}
+	
+	private static void validateFieldValueTypesForObject(Object obj, Set<Object> previouslyValidatedObjects) throws IllegalArgumentException, IllegalAccessException, SerializationException {
+		if(obj == null) {
+			return;
+		}
+		
+		Field[] fields = obj.getClass().getDeclaredFields();
+		if(fields != null && fields.length > 0) {

```

Comment:
fyi `fields.length > 0` is unnecessary, you can for-each loop an empty array without harm